src/test/java/org/apache/commons/validator/routines/VATINValidatorTest.java

```diff
@@ -0,0 +1,251 @@
+            "SI21649405",     // STANOVANJSKO PODJETJE KONJICE D.O.O.
+            "SK2120567108",   // Markon s.r.o. Bratislava
+            "XI366303068",    // donnellygroup.co.uk, old style cd
+            "XI434031494",
+            "XI110305878",    // bullseyecountrysport.co.uk, new style
+            "XI174918964",
+            "XI174918964001", // with 3 digits for branch
+    };
+    // @formatter:on
+
+    // @formatter:off
+    private static final String[] INVALID_VATIN_FIXTURES = {
+            "",                        // empty
+            "   ",                     // empty
+            "A",                       // too short
+            "AB",                      // too short
+            "AT99999999",     // too short, missing U:ATU
+            "ATu99999999",    // lowercase U:ATU
+            "ATU9999999",     // too short
+            "BE136695962",    // aus http://www.pruefziffernberechnung.de/U/USt-IdNr.shtml#PZBE (altes 9-stelliges Format)
+            "CY61234567I",    // Must not start with 2,6,7,8
+            "DE00000000",     // too short
+            "EL94327684",     // zu kurz, führende Null fehlt, korrekt: EL094327684
+            "GB110305878",    // bullseyecountrysport.co.uk is from northern ireland
+            "GR023456780",    // falscher Prefix, sollte EL sein!
+            "ESA10215",       // too short
+            "LT100008668621", // C11 is not 1
+            "LV18097230924",  // invalid century
+            "NL004495445B00", // Suffix "00"
+            "RO027825131",    // Must not start with 0
+            "SE136695975523", // Must not end with 23
+            "SI00000019",     // Must not start with 0
+            "SK0000000011",   // Must not start with 0
+            "SK1111111111",   // 3rd digit must not be 1
+            "MC37000005990",  // No CheckDigit routine for Monaco
+    };
+    // @formatter:on
+
+    @Test
+    public void testGetRegexValidatortPatterns() {
+        assertNotNull(VALIDATOR.getValidator("DE").getRegexValidator().getPatterns(), "DE");
+    }
+
+    @Test
+    public void testGetValidator() {
+        assertNotNull(VALIDATOR.getValidator("FI"), "FI");
+        assertNull(VALIDATOR.getValidator("fi"), "fi");
+    }
+
+    @Test
+    public void testHasValidator() {
+        assertTrue(VALIDATOR.hasValidator("FI"), "FI");
+        assertFalse(VALIDATOR.hasValidator("fi"), "fi");
+    }
+
+    @Test
+    public void testInValid() {
+        for (final String f : INVALID_VATIN_FIXTURES) {
+            assertFalse(VALIDATOR.isValid(f), f);
+        }
+    }
+
+    @Test
+    public void testNull() {
+        assertFalse(VALIDATOR.isValid(null), "isValid(null)");
+    }
+
+    @Test
+    public void testSetDefaultValidator1() {
+        final IllegalStateException thrown = assertThrows(IllegalStateException.class, () -> VALIDATOR.setValidator("GB", 15, "GB", null));
+        assertThat(thrown.getMessage(), is(equalTo("The singleton validator cannot be modified")));
```

Comment:
Please, not Hamcrest, not AssertJ, Java is verbose enough as it is without cluttering tests which should be as simple as possible to understand since they also serve to some extent as examples. This is so much simpler to read and maintain:
```
 assertEquals("The singleton validator cannot be modified", thrown.getMessage());
 ```
There is absolutely no need for 2 extra method calls to assert this outcome.